CONTRIBUTING.md

```diff
@@ -55,4 +69,157 @@ We prefer all communications to be in English.
\ No newline at end of file
+<!-- END MICROSOFT SECURITY.MD BLOCK -->
+
+# Develop for CBL-Mariner
+## Choose a repo
+The first step in developing with CBL-Mariner is to determine which repo fits your use case. If you are just looking to develop using CBL-Mariner's tools and packages, check out the [CBL-MarinerTutorials](https://github.com/microsoft/CBL-MarinerTutorials) repo. `CBL-MarinerTutorials` is geared towards creating custom CBL-Mariner builds rather than contributing to the tools or common packages. 
+```mermaid
+---
+title: Repo decision tree
+---
+flowchart TD
+    %% node styles
+    classDef nodeBig fill:#007FFF,stroke:#333,stroke-width:2px,color:#fff;
+    classDef nodeLittle fill:#007FFF,stroke:#333,stroke-width:2px,color:#fff
+
+    %% nodes
+    id1[Do you want to customize Mariner or contribute to Mariner?]:::nodeBig
+    id2A[Do you want to build locally with Mariner?]:::nodeLittle
+    id2B[Do you want to add an unsupported package? \n either a package Mariner has never supported \nor an updated major/minor version of a package Mariner supports]:::nodeLittle
+    id2C[Do you want to use Mariner for your project? \n ex. bare metal, IoT, embedded devices, etc.]:::nodeLittle
+    id2D([<a href='https://github.com/microsoft/CBL-MarinerTutorials' style='color: #ADD8E6'>Use the CBL-MarinerTutorials repo</a>]):::nodeBig
+    id3A[Do you want to fix an issue in Mariner?]:::nodeLittle
+    id3B[Do you want to add a common package? \n either a package supported by another major distro \nor a package used widely across popular open-source projects]:::nodeLittle
+    id3C[Do you want to modify a supported Mariner package?]:::nodeLittle
+    id3D[Do you want to rebuild Mariner from end-to-end?]:::nodeLittle
+    id3E([<a href='https://github.com/microsoft/CBL-Mariner' style='color: #ADD8E6'>Use the CBL-Mariner repo</a>]):::nodeBig
+
+    %% node flow
+    id1 -->|customize| id2A
+    id2A -.-|or| id2B
+    id2B -.-|or| id2C
+    id2C -->|yes to any of the above| id2D
+    id1 --> |contribute|id3A
+    id3A -.-|or| id3B
+    id3B -.-|or| id3C
+    id3C -.-|or| id3D
+    id3D-->|yes to any of the above|id3E
+```
+## New packages
+New packages fall into one of three buckets: Mariner `SPECS`, Mariner `SPECS-EXTENDED`, or outside support. Mariner core (`SPECS`) includes full support and coverage. Packages in `SPECS-EXTENDED` are only guaranteed to compile successfully. Outside support means the package is not built or published by Mariner but instead by the user. Please see [CBL-MarinerTutorials](https://github.com/microsoft/CBL-MarinerTutorials) for building outside packages.
+
+| Package Support Level | Builds | Published | CVE SLA | Restrictions 
+|:----------------------|:-------|:----------|:--------|:------------
+|SPECS                  |**Yes** |**Yes**    |**Yes**  | - have a viable upstream source which actively addresses CVEs <br> - not include project specific code <br> - have value for multiple use cases
```

Comment:
The restrictions seem odd to me reading from an external contributor's perspective. How is a contributor to know when to place their changes in SPECS vs SPECS-EXTENDED?

Perhaps highlight the graduation model, where package contributions first go to SPECS-EXTENDED to be built and iterated on, with a possible path to graduating into SPECS.